proposals/lip-0047.md

```diff
@@ -511,21 +647,29 @@ chainProperties.roundEndHeight = g.header.height + len(snapshotStore(0).addresse
 BFTThreshold = snapshotStore(0).threshold
 setBFTParameters(BFTThreshold, BFTThreshold, snapshotStore(0).validators)
 
+# Pass the BLS keys and generator keys to the Validators module
+for every validator element in genesisBlockAssetObject.validatorSubstore.validators:
+    registerValidatorKeys(validator.address, validator.proofOfPossession, validator.generatorKey, validator.blsKey)
```

Comment:
The return value should be checked, and block execution cancelled if the the return value is not `True`.

Reply:
I think like this it is still not very clear as line 648 is just a comparison statement. But it's not clear that something fails here if `registerValidatorKeys` returns `False`. I would propose something like:

``` 
for every validator element in genesisBlockAssetObject.validatorSubstore.validators:
    if registerValidatorKeys(validator.address, validator.proofOfPossession, validator.generatorKey, validator.blsKey) != True:
        discard the block 
```

Maybe this functions becomes easier to understand if the logic is put into a function that returns True or False, and in the introduction you saysthat the block is discarded if the function returns `False`.

Reply:
I think it is already clear that if this function does not return true, the processing must stop. 

Anyway, I went for your first suggestion here